evm/hyperliquid-block-helper.js

```diff
@@ -0,0 +1,182 @@
+const { addEvmOptions } = require('./cli-utils');
+const execAsync = promisify(exec);
+
+function addressToBytes(address) {
+    return Buffer.from(address.replace('0x', ''), 'hex');
+}
+
+function actionHash(action, vaultAddress, nonce) {
+    const msgpack = require('msgpack-lite');
+    const actionData = msgpack.encode(action);
+    const nonceBuffer = Buffer.alloc(8);
+    nonceBuffer.writeBigUInt64BE(BigInt(nonce));
+
+    let vaultBuffer;
+    if (vaultAddress === null || vaultAddress === undefined) {
+        vaultBuffer = Buffer.from([0x00]);
+    } else {
+        const addressBytes = addressToBytes(vaultAddress);
+        vaultBuffer = Buffer.concat([Buffer.from([0x01]), addressBytes]);
+    }
+
+    const data = Buffer.concat([actionData, nonceBuffer, vaultBuffer]);
+    const { keccak256 } = require('ethers/lib/utils');
+    return keccak256(data);
+}
+
+function constructPhantomAgent(hash, isMainnet) {
+    return {
+        source: isMainnet ? 'a' : 'b',
+        connectionId: hash,
+    };
+}
+
+async function signL1Action(wallet, action, activePool, nonce, isMainnet) {
+    const hash = actionHash(action, activePool, nonce);
+    const phantomAgent = constructPhantomAgent(hash, isMainnet);
+
+    const domain = {
+        chainId: 1337,
+        name: 'Exchange',
+        verifyingContract: '0x0000000000000000000000000000000000000000',
+        version: '1',
+    };
+
+    const types = {
+        Agent: [
+            { name: 'source', type: 'string' },
+            { name: 'connectionId', type: 'bytes32' },
+        ],
+        EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+        ],
+    };
+
+    const signature = await wallet._signTypedData(domain, { Agent: types.Agent }, phantomAgent);
+    const { ethers } = require('ethers');
+    const sig = ethers.utils.splitSignature(signature);
+
+    return { r: sig.r, s: sig.s, v: sig.v };
+}
+
+async function sendRequest(action, signature, nonce, isMainnet) {
+    const payload = { action, signature, nonce };
+    const endpoint = isMainnet ? 'https://api.hyperliquid.xyz/exchange' : 'https://api.hyperliquid-testnet.xyz/exchange';
+
+    const curlCommand = `curl -s -X POST "${endpoint}" \
+        -H "Content-Type: application/json" \
+        -H "User-Agent: Mozilla/5.0 (compatible; Hyperliquid-Block-Helper/1.0)" \
+        -d '${JSON.stringify(payload)}' \
+        --connect-timeout 15 \
+        --max-time 30`;
+
+    const { stdout, stderr } = await execAsync(curlCommand);
+
+    if (stderr && !stderr.includes('curl')) {
+        throw new Error(`curl stderr: ${stderr}`);
+    }
+
+    const result = JSON.parse(stdout);
+    return result;
+}
+
+async function switchBlockSize(privateKey, useBig, network = 'mainnet') {
+    validateParameters({ isNonEmptyString: { privateKey } });
```

Comment:
```suggestion
    validateParameters({ isEVMPrivateKey: { privateKey } });
```
add a new validator instead?